### PR TITLE
Add Garfield++

### PR DESCRIPTION
**Pull request by mirguest**
BEGINRELEASENOTES
- New package Garfield++.

ENDRELEASENOTES

### CODE REVIEW

**Comment by mirguest** [comment]
Hi @vvolkl ,
Due to CEPCSW will depend on the Garfield++, I create a new receipt for Garfield++. Could you review this PR?
Thank you!
Tao

File: var/spack/repos/builtin/packages/garfieldpp/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class Garfieldpp(CMakePackage):
+    """Garfield++ is a toolkit for the detailed simulation of particle
+       detectors based on ionisation measurement in gases and semiconductors. """
+
+    homepage = "https://garfieldpp.web.cern.ch/garfieldpp/"
+    url      = "https://gitlab.cern.ch/garfield/garfieldpp/-/archive/4.0/garfieldpp-4.0.tar.gz"
+    git      = "https://gitlab.cern.ch/garfield/garfieldpp.git"
+
+    maintainers = ['mirguest']
+
```
Comment (vvolkl) [inline var/spack/repos/builtin/packages/garfieldpp/package.py:21]:
```suggestion

    tags = ['hep']

    maintainers = ['mirguest']

```

**Comment by vvolkl** [comment]
Hi @mirguest,

sure, only comment I have is to add the `hep` tag which is useful to find available packages in spack. I see you added it in upstream spack already too: For any package recipe that are merged there, you can just let me know and  I can cherrypick the commits from there, no need for a PR.
Cheers,
Valentin

**Comment by mirguest** [comment]
Hi @vvolkl ,
Thanks for the comment. 
OK, I will let you know next time. Thank you. 
Tao

**Pull request merged**